ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager2.java

```diff
@@ -3816,4 +3816,29 @@ private void testDropMaterializedView(boolean blocking) throws Exception {
     Assert.assertEquals("Expecting 2 rows and found " + res.size(), 2, res.size());
     driver.run("drop materialized view mv_tab_acid");
   }
+
+  @Test
+  public void testAddDropConstraintNonBlocking() throws Exception {
+    HiveConf.setBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED, true);
+    dropTable(new String[] {"tab_acid"});
+
+    driver.run("create table if not exists tab_acid (a int, b int) " +
+      "stored as orc TBLPROPERTIES ('transactional'='true')");
+    driver.run("insert into tab_acid (a,b) values(1,2),(3,4)");
+
+    driver.compileAndRespond("alter table tab_acid ADD CONSTRAINT a_PK PRIMARY KEY (`a`) DISABLE NOVALIDATE");
+    driver.lockAndRespond();
+    
+    List<ShowLocksResponseElement> locks = getLocks();
+    checkLock(LockType.EXCL_WRITE,
+      LockState.ACQUIRED, "default", "tab_acid", null, locks);
+    driver.close();
+    
+    driver.compileAndRespond("alter table tab_acid  DROP CONSTRAINT a_PK");
+    driver.lockAndRespond();
+
+    locks = getLocks();
+    checkLock(LockType.EXCL_WRITE,
+      LockState.ACQUIRED, "default", "tab_acid", null, locks);
+  }
```

Comment:
It could be separated to two tests, but no strong feelings about it, even the merged test case is small enough.